ShipMonitor/ConfigurationWindow.xaml.cs

```diff
@@ -144,26 +141,19 @@ private void shipsUpdated(object sender, SelectionChangedEventArgs e)
                 shipMonitor()?.Save();
             }
         }
-    }
-
-    public class ValidIpaRule : ValidationRule
-    {
-        private static readonly Regex ipaRegex = new Regex(@"^[bdfɡhjklmnprstvwzxaɪ˜iu\.ᵻᵿɑɐɒæɓʙβɔɕçɗɖðʤəɘɚɛɜɝɞɟʄɡ(ɠɢʛɦɧħɥʜɨɪʝɭɬɫɮʟɱɯɰŋɳɲɴøɵɸθœɶʘɹɺɾɻʀʁɽʂʃʈʧʉʊʋⱱʌɣɤʍχʎʏʑʐʒʔʡʕʢǀǁǂǃˈˌːˑʼʴʰʱʲʷˠˤ˞n̥d̥ŋ̊b̤a̤t̪d̪s̬t̬b̰a̰t̺d̺t̼d̼t̻d̻t̚ɔ̹ẽɔ̜u̟e̠ël̴n̴ɫe̽e̝ɹ̝m̩n̩l̩e̞β̞e̯e̘e̙ĕe̋éēèȅx͜xx͡x↓↑→↗↘]+$");
 
-        public override ValidationResult Validate(object value, CultureInfo cultureInfo)
+        // Fixup IPA text by replacing spaces
+        private void PhoneticName_TextChanged(object sender, TextChangedEventArgs e)
         {
-            if (value == null)
-            {
-                return ValidationResult.ValidResult;
-            }
-            string val = value.ToString();
-            if (ipaRegex.Match(val).Success)
+            if (sender is TextBox textBox)
             {
-                return ValidationResult.ValidResult;
-            }
-            else
-            {
-                return new ValidationResult(false, "Invalid IPA");
+                if (textBox.IsLoaded)
+                {
+                    // Replace any spaces, maintaining the original caret position
+                    int caretIndex = textBox.CaretIndex;
+                    textBox.Text = textBox.Text.Replace(" ", "ˈ");
```

Comment:
This may look a little odd if there are spaces to the left of the caret, but the fix (split the string at at caret position, trim both parts separately, then set the caret to the new length of the first half) is probably not worth it.

Reply:
It's a 1:1 replacement. The overall length of the string will not change and consequently I'm favoring the simpler approach. :-)

Reply:
My mistake. I misread the code.